community/graphdb-api/src/main/java/org/neo4j/graphdb/GraphDatabaseService.java

```diff
@@ -238,6 +238,23 @@
     Transaction beginTx();
 
     /**
+     * Starts a new {@link Transaction transaction} with custom timeout and associates it with the current thread.
+     * Timeout will be taken into account <b>only</b> when execution guard is enabled.
+     * <p>
+     * <em>All database operations must be wrapped in a transaction.</em>
+     * <p>
+     * If you attempt to access the graph outside of a transaction, those operations will throw
+     * {@link NotInTransactionException}.
+     * <p>
+     * Please ensure that any returned {@link ResourceIterable} is closed correctly and as soon as possible
+     * inside your transaction to avoid potential blocking of write operations.
+     *
+     * @param timeout transaction timeout
+     * @return a new transaction instance
+     */
+    Transaction beginTx( long timeout );
```

Comment:
sounds like a good idea since then users do not need to perform conversion themself and that will be more readable